pkg/doc/verifiable/credential_jwt.go

```diff
@@ -52,107 +53,78 @@ func newJWTCredClaims(vc *Credential, minimizeVc bool) (*JWTCredClaims, error) {
 		raw = vc.raw()
 	}
 
-	credClaims := &JWTCredClaims{
-		Claims:     jwtClaims,
-		Credential: raw,
-	}
-
-	return credClaims, nil
-}
-
-// jwtVCClaim is used to get content of "vc" claim of JWT.
-type jwtVCClaim struct {
-	VC map[string]interface{} `json:"vc,omitempty"`
-}
-
-func mergeRefinedVC(jsonCred map[string]interface{}, rawCred *rawCredential) error {
-	rawData, err := json.Marshal(rawCred)
-	if err != nil {
-		return err
-	}
-
-	var rawMap map[string]interface{}
-
-	err = json.Unmarshal(rawData, &rawMap)
+	vcMap, err := raw.asMap()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// make the merge
-	for k, v := range rawMap {
-		jsonCred[k] = v
+	credClaims := &JWTCredClaims{
+		Claims: jwtClaims,
+		VC:     vcMap,
 	}
-
-	return nil
-}
-
-// jwtCredClaimsDecoder parses JWT claims from serialized token into JWTCredClaims struct and JSON object.
-// The implementation depends on the type of serialization, e.g. signed (JWT), unsecured (plain JWT).
-type jwtCredClaimsDecoder interface {
-	UnmarshalClaims(rawJwt []byte) (*JWTCredClaims, error)
-	UnmarshalVCClaim(rawJwt []byte) (map[string]interface{}, error)
+	return credClaims, nil
 }
 
-// decodeCredJWT parses JWT from the specified bytes array in compact format using jwtDecoder.
-// It returns decoded Verifiable Credential refined by JWT Claims in raw byte array and rawCredential form.
-func decodeCredJWT(rawJWT []byte, credClaimsDecoder jwtCredClaimsDecoder) ([]byte, *rawCredential, error) {
-	credClaims, err := credClaimsDecoder.UnmarshalClaims(rawJWT)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to decode Verifiable Credential JWT claims: %w", err)
-	}
+// JWTCredClaimsUnmarshaller unmarshals verifiable credential bytes into JWT claims with extra "vc" claim.
+type JWTCredClaimsUnmarshaller func(vcJWTBytes []byte) (*JWTCredClaims, error)
 
-	credJSON, err := credClaimsDecoder.UnmarshalVCClaim(rawJWT)
+// decodeCredJWT parses JWT from the specified bytes array in compact format using unmarshaller.
+// It returns decoded Verifiable Credential refined by JWT Claims in raw byte array form.
+func decodeCredJWT(rawJWT []byte, unmarshaller JWTCredClaimsUnmarshaller) ([]byte, error) {
+	credClaims, err := unmarshaller(rawJWT)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to decode raw Verifiable Credential JWT claims: %w", err)
+		return nil, fmt.Errorf("failed to decode Verifiable Credential JWT claims: %w", err)
```

Comment:
Wrapping messages should be kept succinct. https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

Reply:
Agree! I think it deserves the creation of a separate issue, to clean-up the whole `credential` package.
#570